src/Protocol/Protocol.h

```diff
@@ -56,9 +56,84 @@ class cProtocol
 
 	virtual ~cProtocol() {}
 
+	/** A list of all outgoing packets */
+	enum eOutgoingPackets
+	{
+		sendAttachEntity = 0,
+		sendBlockAction,
+		sendBlockBreakAnim,
+		sendBlockChange,
+		sendBlockChanges,
+		sendCameraSetTo,
+		sendChatRaw,
+		sendCollectEntity,
+		sendDestroyEntity,
+		sendDifficulty,
+		sendDisconnectDuringLogin,
+		sendDisconnectDuringGame,
+		sendDisplayObjective,
+		sendEditSign,
+		sendEntityAnimation,
+		sendEntityEffect,
+		sendEntityEquipment,
+		sendEntityHeadLook,
+		sendEntityLook,
+		sendEntityMeta,
+		sendEntityProperties,
+		sendEntityRelMove,
+		sendEntityRelMoveLook,
+		sendEntityStatus,
+		sendEntityVelocity,
+		sendExperience,
+		sendExperienceOrb,
+		sendExplosion,
+		sendGameMode,
+		sendHealth,
+		sendInventorySlot,
+		sendJoinGame,
+		sendKeepAlive,
+		sendLeashEntity,
+		sendMapData,
+		sendPaintingSpawn,
+		sendParticleEffect,
+		sendPlayerAbilities,
+		sendPlayerList,
+		sendPlayerMaxSpeed,
+		sendPlayerMoveLook,
+		sendPlayerSpawn,
+		sendPluginMessage,
+		sendRemoveEntityEffect,
+		sendRespawn,
+		sendScoreboardObjective,
+		sendScoreUpdate,
+		sendSpawnObject,
+		sendSoundEffect,
+		sendSoundParticleEffect,
+		sendSpawnGlobalEntity,
+		sendSpawnMob,
+		sendSpawnPosition,
+		sendStatistics,
+		sendTabCompletion,
+		sendTeleportEntity,
+		sendTimeUpdate,
+		sendTitle,
+		sendUnloadChunk,
+		sendUpdateBlockEntity,
+		sendUpdateSign,
+		sendUseBed,
+		sendWeather,
+		sendWindowItems,
+		sendWindowClose,
+		sendWindowOpen,
+		sendWindowProperty
+	};
+
 	/** Called when client sends some data */
 	virtual void DataReceived(const char * a_Data, size_t a_Size) = 0;
 
+	/** Called to figure out a packet ID */
```

Comment:
This could use a bit more description.
`/** Called to figure out the actual packet ID from the protocol-agnostic packet type. */

Reply:
Fixed.

Reply:
Not really, you swapped the `protocol-agnostic packet type` into `packet agnostic protocol type` :)

Reply:
And here I was, thinking all P-words were interchangeable :-P